packages/video_player_avplayer/lib/video_player.dart

```diff
@@ -0,0 +1,1224 @@
+  final String? errorDescription;
+
+  /// The [size] of the currently loaded video.
+  final Size size;
+
+  /// Indicates whether or not the video has been loaded and is ready to play.
+  final bool isInitialized;
+
+  /// Indicates whether or not the video is in an error state. If this is true
+  /// [errorDescription] should have information about the problem.
+  bool get hasError => errorDescription != null;
+
+  /// Returns [size.width] / [size.height].
+  ///
+  /// Will return `1.0` if:
+  /// * [isInitialized] is `false`
+  /// * [size.width], or [size.height] is equal to `0.0`
+  /// * aspect ratio would be less than or equal to `0.0`
+  double get aspectRatio {
+    if (!isInitialized || size.width == 0 || size.height == 0) {
+      return 1.0;
+    }
+    final double aspectRatio = size.width / size.height;
+    if (aspectRatio <= 0) {
+      return 1.0;
+    }
+    return aspectRatio;
+  }
+
+  /// Returns a new instance that has the same values as this current instance,
+  /// except for any overrides passed in as arguments to [copyWidth].
+  VideoPlayerValue copyWith({
+    Duration? duration,
+    Size? size,
+    Duration? position,
+    Caption? caption,
+    Duration? captionOffset,
+    List<Track>? tracks,
+    int? buffered,
+    bool? isInitialized,
+    bool? isPlaying,
+    bool? isLooping,
+    bool? isBuffering,
+    double? volume,
+    double? playbackSpeed,
+    String? errorDescription = _defaultErrorDescription,
+  }) {
+    return VideoPlayerValue(
+      duration: duration ?? this.duration,
+      size: size ?? this.size,
+      position: position ?? this.position,
+      caption: caption ?? this.caption,
+      captionOffset: captionOffset ?? this.captionOffset,
+      tracks: tracks ?? this.tracks,
+      buffered: buffered ?? this.buffered,
+      isInitialized: isInitialized ?? this.isInitialized,
+      isPlaying: isPlaying ?? this.isPlaying,
+      isLooping: isLooping ?? this.isLooping,
+      isBuffering: isBuffering ?? this.isBuffering,
+      volume: volume ?? this.volume,
+      playbackSpeed: playbackSpeed ?? this.playbackSpeed,
+      errorDescription: errorDescription != _defaultErrorDescription
+          ? errorDescription
+          : this.errorDescription,
+    );
+  }
+
+  @override
+  String toString() {
+    return '${objectRuntimeType(this, 'VideoPlayerValue')}('
+        'duration: $duration, '
+        'size: $size, '
+        'position: $position, '
+        'caption: $caption, '
+        'captionOffset: $captionOffset, '
+        'tracks: $tracks, '
+        'buffered: $buffered, '
+        'isInitialized: $isInitialized, '
+        'isPlaying: $isPlaying, '
+        'isLooping: $isLooping, '
+        'isBuffering: $isBuffering, '
+        'volume: $volume, '
+        'playbackSpeed: $playbackSpeed, '
+        'errorDescription: $errorDescription)';
+  }
+}
+
+typedef _InitDartApi = int Function(Pointer<Void>);
+typedef _InitDartApiNative = IntPtr Function(Pointer<Void>);
```

Comment:
I remove FFI codes since it will lock the platform thread, and use method channel to implement request license via dart callback. 
[Remove FFI codes and implement request license through method channel (](https://github.com/flutter-tizen/plugins/pull/620/commits/7e90b1b1a4b645f69e6266a85ee54ef0404b8a14)https://github.com/xiaowei-guan/plugins/pull/7